geode-core/src/test/java/org/apache/geode/management/internal/cli/commands/ListAsyncEventQueuesCommandDUnitTest.java

```diff
@@ -0,0 +1,81 @@
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.management.internal.cli.commands;
+
+import static org.apache.geode.test.junit.rules.GfshCommandRule.PortType.jmxManager;
+
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import org.apache.geode.internal.cache.wan.MyAsyncEventListener;
+import org.apache.geode.test.dunit.rules.LocatorServerStartupRule;
+import org.apache.geode.test.dunit.rules.MemberVM;
+import org.apache.geode.test.junit.categories.UnitTest;
+import org.apache.geode.test.junit.rules.GfshCommandRule;
+
+
+@Category(UnitTest.class)
+public class ListAsyncEventQueuesCommandDUnitTest {
+
+  @ClassRule
+  public static LocatorServerStartupRule lsRule = new LocatorServerStartupRule();
+
+  @Rule
+  public GfshCommandRule gfsh = new GfshCommandRule(locator::getJmxPort, jmxManager);
+
+  private static MemberVM locator;
+
+  @BeforeClass
+  public static void beforeClass() throws Exception {
+    locator = lsRule.startLocatorVM(0);
+    lsRule.startServerVM(1, "group1", locator.getPort());
+    lsRule.startServerVM(2, "group2", locator.getPort());
+  }
+
+  @Test
+  public void list() throws Exception {
+    gfsh.executeAndAssertThat("list async-event-queue").statusIsSuccess()
+        .containsOutput("No Async Event Queues Found");
+
+    gfsh.executeAndAssertThat("create async-event-queue --id=queue1 --group=group1 --listener="
+        + MyAsyncEventListener.class.getName()).statusIsSuccess();
+
+    gfsh.executeAndAssertThat("create async-event-queue --id=queue2 --group=group2 --listener="
+        + MyAsyncEventListener.class.getName()).statusIsSuccess();
+
+    locator.waitTillAsyncEventQueuesAreReadyOnServers("queue1", 1);
+    locator.waitTillAsyncEventQueuesAreReadyOnServers("queue2", 1);
+
+    gfsh.executeAndAssertThat("list async-event-queue").statusIsSuccess()
+        .tableHasRowCount("Member", 2).tableHasRowWithValues("Member", "ID", "server-1", "queue1")
+        .tableHasRowWithValues("Member", "ID", "server-2", "queue2");
+
+    // create another async event queue on the entire cluster, verify that the command will list all
+    gfsh.executeAndAssertThat(
+        "create async-event-queue --id=queue --listener=" + MyAsyncEventListener.class.getName())
+        .statusIsSuccess();
+
+    gfsh.executeAndAssertThat("list async-event-queue").statusIsSuccess()
+        .tableHasRowCount("Member", 4).tableHasRowWithValues("Member", "ID", "server-1", "queue1")
+        .tableHasRowWithValues("Member", "ID", "server-2", "queue2")
+        .tableHasRowWithValues("Member", "ID", "server-1", "queue")
+        .tableHasRowWithValues("Member", "ID", "server-2", "queue");
```

Comment:
Point of discussion (not necessary to change, but something to consider) : the signature of {{tableHasRowWithValues}} has {{headersThenValues}}. I didn't notice this from the previous change to CommandResultAssert, but I think that that ordering of the {{String}} arguments as {{header, value}} pairs is more readable. With only one or two values in the row it's not too confusing, but with a longer list of values of interest, then associating the matching header and values becomes less clear when reviewing code that calls the method.

Reply:
so you are suggesting the caller should call something like: tableHasRowWithValues("Member,server-2", "ID,queue2")

this would make the processing a bit harder, but still doable. It's just the way these GFJsonObject are constructed, We will have to refactor out this entire result object anyway, good things we are lumping all the processing in this CommandResultAssert object so it's one spot to change.

Reply:
No, what I would envision is tableHasRowWithValues(Member, server-2, ID, queue2). I would keep the arguments separate but order arguments as pairs ..., header, value, ... . I agree that processing this wouldn't be quite as straightforward as 

    String[] headers = Arrays.copyOfRange(headersThenValues, 0, numberOfColumn);
    String[] expectedValues =
        Arrays.copyOfRange(headersThenValues, numberOfColumn, headersThenValues.length);